src/libraries/Microsoft.PowerFx.Core/Entities/External/InternalTableMetadata.cs

```diff
@@ -0,0 +1,144 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.PowerFx.Core.App;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core.Entities
+{
+    internal class InternalTableMetadata : IExternalTableMetadata
+    {        
+        private readonly AggregateType _type;
```

Comment:
>AggregateType [](http://example.com/codeflow?start=25&length=13)

should this be the new derived type so we can call the new virtuals?

Reply:
I don't understand what you mean

Reply:
This is the TabularRecordType, yes